python/ngen_conf/src/ngen/config/init_config/cfe.py

```diff
@@ -0,0 +1,165 @@
+    # maximum soil water storage (smcmax * depth) for the initial timestep
+    soil_storage: FloatUnitPair[m_per_m]  # 66.7%
+    # number of Nash lf reservoirs (optional, defaults to 2, ignored if storage values present)
+    k_nash: FloatUnitPair[empty]
+    # Nash Config param - primary reservoir
+    k_lf: FloatUnitPair[empty]
+    # Nash Config param - secondary reservoir
+    nash_storage: List[float]
+    # Giuh ordinates in dt time steps
+    giuh_ordinates: List[float]
+    # set to `1` if `forcing_file=BMI`
+    num_timesteps: int = Field(1, gte=0)
+
+    # prints various debug and bmi info
+    verbosity: int = Field(0, gte=0, lte=3)
+
+    # direct runoff
+    surface_partitioning_scheme: Literal["Schaake", "Xinanjiang"]
+
+    # TODO: add support for below fields
+
+    # # when `CFE coupled to SoilMoistureProfile`
+    # aet_rootzone: bool  # True, true, 1
+    # # layer of the soil that is the maximum root zone depth. That is, the depth of the layer where the AET is drawn from
+    # max_root_zone_layer: FloatUnitPair[m]
+    # # an array of depths from the surface. Example, soil_layer_depths=0.1,0.4,1.0,2.0
+    # soil_layer_depths: List[float]
+    # # `ice-fraction based runoff` | when `CFE coupled to SoilFreezeThaw`
+    # sft_coupled: bool  # True, true, 1
+
+    _coerce_lists = validator(
+        "nash_storage",
+        "giuh_ordinates",
+        allow_reuse=True,
+        pre=True,
+    )(validators.str_split(",", strip=True))
+
+    class Config(serde.IniSerializerDeserializer.Config):
+        def _serialize_list(l: List[float]) -> str:
+            return ",".join(map(lambda x: str(x), l))
+
+        field_serializers = {
+            "nash_storage": _serialize_list,
+            "giuh_ordinates": _serialize_list,
+        }
+
+        fields = {
+            "cgw": {"alias": "Cgw"},
+            "k_lf": {"alias": "K_lf"},
+            "k_nash": {"alias": "K_nash"},
+            "soil_params_depth": {"alias": "soil_params.depth"},
+            "soil_params_b": {"alias": "soil_params.b"},
+            "soil_params_mult": {"alias": "soil_params.mult"},
+            "soil_params_satdk": {"alias": "soil_params.satdk"},
+            "soil_params_satpsi": {"alias": "soil_params.satpsi"},
+            "soil_params_slop": {"alias": "soil_params.slop"},
+            "soil_params_smcmax": {"alias": "soil_params.smcmax"},
+            "soil_params_wltsmc": {"alias": "soil_params.wltsmc"},
+            "soil_params_expon": {"alias": "soil_params.expon"},
+            "soil_params_expon_secondary": {"alias": "soil_params.expon_secondary"},
+        }
+
+
+class CFESchaake(CFEBase):
+    surface_partitioning_scheme: Literal["Schaake"] = "Schaake"
+
+
+class CFEXinanjiang(CFEBase):
+    # direct runoff
+    surface_partitioning_scheme: Literal["Xinanjiang"] = "Xinanjiang"
+    a_xinanjiang_inflection_point_parameter: int
+    b_xinanjiang_shape_parameter: int
+    x_xinanjiang_shape_parameter: int
```

Comment:
should the type be `float` here (or doesn't matter)? in cfe these parameters are double

Reply:
Great catch, yes they should be `float`s.